test/session_test.dart

```diff
@@ -279,39 +279,41 @@ Future<void> main([List<String>? args]) async {
   });
 
-  baasTest('SyncSession test error handler', (configuration) async {
+  baasTest('BadFlexibleSyncQueryError test error handler', (configuration) async {
     final app = App(configuration);
     final user = await getIntegrationUser(app);
-    final config = Configuration.flexibleSync(user, [Task.schema], syncErrorHandler: (syncError) {
-      expect(syncError, isA<SyncSessionError>());
-      final sessionError = syncError.as<SyncSessionError>();
-      expect(sessionError.category, SyncErrorCategory.session);
-      expect(sessionError.isFatal, false);
-      expect(sessionError.code, SyncSessionErrorCode.badAuthentication);
-      expect(sessionError.detailedMessage, "Simulated session error");
-      expect(sessionError.message, "Bad user authentication (BIND)");
+    final config = Configuration.flexibleSync(user, syncSchema, syncErrorHandler: (syncError) {
+      expect(syncError, isA<BadFlexibleSyncQueryError>());
+      expect(syncError.message, isNotEmpty);
     });
 
     final realm = getRealm(config);
 
-    realm.syncSession.raiseError(SyncErrorCategory.session, SyncSessionErrorCode.badAuthentication.code, false);
+    realm.syncSession.raiseError(SyncErrorCodes.invalidSubscriptionQuery.code, false);
   });
 
-  baasTest('SyncSession test fatal error handler', (configuration) async {
+  baasTest('UnrecoverableSyncError test error handler', (configuration) async {
     final app = App(configuration);
     final user = await getIntegrationUser(app);
-    final config = Configuration.flexibleSync(user, [Task.schema], syncErrorHandler: (syncError) {
-      expect(syncError, isA<SyncClientError>());
-      final syncClientError = syncError.as<SyncClientError>();
-      expect(syncClientError.category, SyncErrorCategory.client);
-      expect(syncClientError.isFatal, true);
-      expect(syncClientError.code, SyncClientErrorCode.badChangeset);
-      expect(syncClientError.detailedMessage, "Simulated session error");
-      expect(syncClientError.message, "Bad changeset (DOWNLOAD)");
+    final config = Configuration.flexibleSync(user, syncSchema, syncErrorHandler: (syncError) {
+      expect(syncError, isA<UnrecoverableSyncError>());
+      expect(syncError.message, isNotEmpty);
     });
     final realm = getRealm(config);
 
-    realm.syncSession.raiseError(SyncErrorCategory.client, SyncClientErrorCode.badChangeset.code, true);
+    realm.syncSession.raiseError(SyncErrorCodes.syncProtocolInvariantFailed.code, true);
+  });
+
+  baasTest('WrongSyncTypeError test error handler', (configuration) async {
+    final app = App(configuration);
+    final user = await getIntegrationUser(app);
+    final config = Configuration.flexibleSync(user, syncSchema, syncErrorHandler: (syncError) {
+      expect(syncError, isA<WrongSyncTypeError>());
+      expect(syncError.message, isNotEmpty);
+    });
+    final realm = getRealm(config);
+
+    realm.syncSession.raiseError(SyncErrorCodes.wrongSyncType.code, true);
```

Comment:
this does not test if the callback was invoked. Should we have an `expect` that checks the callback was invoked otherwise it will be green when in fact the expects in the callback neve happened

Reply:
ignore. it is our raiseError method so we don't need to verify the callback is invoked.